Kernel/.gitignore

```diff
@@ -1,4 +1,3 @@
 kernel.map
-sync-local.sh
```

Comment:
Maybe we could add a warning in the build script, and remove it in a few months?

Something like:
```sh
if [ -f "$SERENITY_ROOT"/Kernel/sync-local.sh ]; then
    # TODO: Deprecated on 2021-01-30. In a few months, remove this 'if'.
    echo "Warning: $SERENITY_ROOT/Kernel/sync-local.sh should move to $SERENITY_ROOT/sync-local.sh"
fi
```

or the likes.

Reply:
Good call, done! (lots of red text :D)